### DISCUSSION TITLE
ERROR: (gcloud.pubsub.subscriptions.create) Error parsing [subscription]. The [subscription] resource is not properly specified. Failed to find attribute [project]. The attribute can be set in the following ways: - provide the argument `subscription` on the command line with a fully specified name - provide the argument `--project` on the command line - set the property `core/project` -bash: Created: command not found

### DISCUSSION THREAD

**Discussion by NoahC1986** [potmat/homebridge-google-nest-sdm#56] [Q&A]
I have tried to read through the other threads and looked online as much as I can. I am very new to all of this so please forgive me. I have been trying to follow the subscribe to events step and can't get back the create a pull subscription. The gcp-project-name is the same as the project ID from the google cloud correct ? The project ID is the same as the SDM project ID from the UI entries as well? And what do I need to put for the subscription name. It makes it seem like you can put whatever you like but it also mentions it needs to have the topic ID as well? 

I also tried utilizing the pub/sub area to create a subscription but that still gives me a error for initialization in homebridge. Again I know from what ive read through the other issues that you don't support google issues but id really appreciate the help.

**Comment by potmat** [comment]
Hi @NoahC1986, did you follow the advice in the FAQ first? 

"The Nest plugin for Home Assistant uses much the same process (don't forget the ["ONE IMPORTANT DIFFERENCE"](https://github.com/potmat/homebridge-google-nest-sdm#where-do-the-config-values-come-from) section above). It has an [illustrated guide](https://www.home-assistant.io/integrations/nest/) that you may find helpful. You can also try reaching out to others on [Discord](https://discord.gg/kqNCe2D) or [Reddit](https://www.reddit.com/r/homebridge/), some people there may be able to help.

**Comment by NoahC1986** [comment]
I tried to follow each step as best as I could.

**Comment by orangewinger** [comment]
I had this problem today, If you're using the GCloud Shell then make sure you select the correct project in the droplist at the top as that will automatically set the project id for you in a new shell.

**Comment by potmat** [comment]
@NoahC1986: Did this resolve your issue?

**Comment by potmat** [comment]
Converting to a discussion.

**Comment by qCzar** [comment]
I've having this issue as well, at least I think it's the same issue.

https://cloud.google.com/pubsub/docs/create-topic#resource_names

I'm stuck here as Google nor the Readme for this really go into depth what the "ID" is. Google's KB makes it seem like I can pick any name I want, but I tried that and got an error. I can't find anything in the Cloud Console that would indicate it's a different ID than I've used before. I tried using the Project ID again but it also failed

EDIT: Just kidding, you need to create a topic at the step below in the link. I don't get an error when the plugin boots but I still don't have video. But I think my similarities with this issue are no more. On to searching for other discussions or issues.

**Reply by mkormendy** [reply]
Following the steps, when creating a pull subscription in the Cloud Shell Terminal , just name the subscription-id anything meaningful to you. I used something like "nest-pubsub-subscription".
